### ISSUE TITLE
Sources that do a lot of processing might hang up (e.g. MangaSee)

### ISSUE THREAD

**Issue by hanatsumi** [hanatsumi/rakuyomi#55]
**Describe the bug**

Some sources that do a lot of processing inside the extension might hang up rakuyomi when used. An example is the MangaSee source, which [parses a 3MB JSON string](https://github.com/Skittyblock/aidoku-community-sources/blob/38ee8754a4c5f4cded68dbede63cc995c08ec6a7/src/rust/en.nepnep/src/lib.rs#L41).

**To Reproduce**

1. Attempt to do any search with the MangaSee source.

**Expected behavior**

Not hangup

**Comment by hanatsumi** [comment]
This might be fixable by using the `json::parse` WASM import instead of using `serde_json` inside the extension code, I'll look into it sometime.

**Comment by Imyr** [comment]
Hi, I encountered this exact issue on the Kobo Libra Colour.

If you allow, I'd like to go through the codebase and try my hand with this and other issues, as I recently came across your repo and have been using it heavily since.

**Comment by hanatsumi** [comment]
@Imyr Sure! Feel free to give it a try. For this specific issue tho, the easiest way to make MangaSee work would probably be to change it upstream, as mentioned in https://github.com/hanatsumi/rakuyomi/issues/55#issuecomment-2451838730

A quick overview of what's happening here: extension/source code is ran inside a WASM interpreter (in our case, [wasmi](https://github.com/wasmi-labs/wasmi)), which, while still reasonably fast for our usecases, isn't fit for the amount of processing that parsing a large JSON document does. If my suspicions are right, using the `json::parse` WASM import would offload most of the work to native code, which should make the extension fast enough to not lock everything up.

For the overarching issue (sources that do a lot of processing might make everything hang up), the solution would be somewhat more complex: we would need to find a way to "kill" the hung-up source, and somehow alert the user that the source has stopped working. As this seems to only be problematic with MangaSee for now, fixing the extension directly seems to be easier.

**Comment by Imyr** [comment]
@hanatsumi Yes, I believe we could set a timeout for each source, inform the user of the problematic source, and show the results from the other sources. This would be better anyway in the long run IMO.
As for the upstream fix, I'll take a look at it.
